cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParser.scala

```diff
@@ -56,6 +57,27 @@ object JSONParser {
     resolved
   }
 
+  def parseLimit(json: JsValue, option: QueryExeOption): (JsValue, QueryExeOption) = {
+    if (option.sliceMills <= 0) { // non-slicing query
+      (json, QueryExeOption(option.sliceMills, option.continueSeconds, None))
+    } else if (option.sliceMills > 0 && (json \ "batch").toOption.isEmpty) { // single slicing query
+      if ((json \\ "limit").isEmpty) {
+        (json, QueryExeOption(option.sliceMills, option.continueSeconds, None))
+      } else {
+        val limit = (json \ "select" \ "limit").as[Int]
+        val updatedSelectJson = (json \ "select").as[JsObject] ++ Json.obj("limit" -> Int.MaxValue)
+        val updatedJson = json.as[JsObject] ++ Json.obj("select" -> updatedSelectJson)
+        (updatedJson, QueryExeOption(option.sliceMills, option.continueSeconds, Some(limit)))
+      }
+    } else {  // batch slicing query
+      if ((json \\ "limit").isEmpty || ((json \\ "limit").nonEmpty && (json \\ "aggregate").nonEmpty)) {
+        (json, QueryExeOption(option.sliceMills, option.continueSeconds, None))
+      } else {
+        throw JsonRequestException("Batch Requests cannot contain \"limit\" field")
```

Comment:
we should let the user know the request is wrong, right? 
but I think it requires a more framework change. let's leave a TODO here.